src/routes/Send.tsx

```diff
@@ -635,15 +660,12 @@ export default function Send() {
                     >
                         <Switch>
                             <Match when={sentDetails()?.failure_reason}>
-                                <MegaEx />
-                                <h1 class="mb-2 mt-4 w-full text-center text-2xl font-semibold md:text-3xl">
-                                    {sentDetails()?.amount
-                                        ? source() === "onchain"
-                                            ? i18n.t("send.payment_initiated")
-                                            : i18n.t("send.payment_sent")
-                                        : sentDetails()?.failure_reason}
-                                </h1>
-                                {/*TODO: add failure hint logic for different failure conditions*/}
+                                <Failure
+                                    reason={
+                                        sentDetails()?.failure_reason ||
+                                        "Payment failed for an unknown reason"
```

Comment:
nit: this `|| string` is redundant because we are checking it is defined in the `Match` if we can remove it it'd be better, if typescript yells at you fine to keep then, safer than putting a `!`

Reply:
yeah solidjs can't do type narrowing for these `if`-style components